base/regex.jl

```diff
@@ -177,9 +177,11 @@ end
 """
     startswith(s::AbstractString, prefix::Regex)
 
-Return `true` if `s` starts with the regex pattern, `prefix`.  Note:
-`occursin` is recommended over the regex version of `startswith` in
-performance critical situations.
+Return `true` if `s` starts with the regex pattern, `prefix`.
+
+!!! note
+    `occursin` is recommended over the regex version of `startswith`
+    in performance critical situations.
```

Comment:
Not sure whether the verbosity is warranted, but we could just explain the internals and let the user decide on tradeoffs:

"""
!!! note 
`startswith` does not compile the anchoring into the regular expression, but instead passes the anchoring as `match_option` to PCRE. If compile time is amortized, `occursin(r"^...", s)` is faster than `startswith(s, r"...")`.
"""